src/Disks/TemporaryFileOnDisk.h

```diff
@@ -16,9 +16,8 @@ using DiskPtr = std::shared_ptr<IDisk>;
 class TemporaryFileOnDisk
 {
 public:
-    explicit TemporaryFileOnDisk(const DiskPtr & disk_);
     explicit TemporaryFileOnDisk(const DiskPtr & disk_, CurrentMetrics::Metric metric_scope);
-    explicit TemporaryFileOnDisk(const DiskPtr & disk_, const String & prefix);
+    explicit TemporaryFileOnDisk(const DiskPtr & disk_, const String & prefix = "tmp");
```

Comment:
Do I understand correctly, it was writing in the wrong place before?

Reply:
Not a place, but filename prefix.
If file is not deleted in destructor (e.g. on hard restart) we will do cleanup on server startup, but check the filename prefix.

https://github.com/ClickHouse/ClickHouse/blob/32b765a4ba577acbfdb09a8d400dad8d4ef0f48d/src/Interpreters/Context.cpp#L865-L871